src/include/btree_inline.h

```diff
@@ -1813,7 +1815,8 @@ __wt_leaf_page_can_split(WT_SESSION_IMPL *session, WT_PAGE *page)
          ins = ins->next[WT_MIN_SPLIT_DEPTH]) {
         count += WT_MIN_SPLIT_MULTIPLIER;
         size += WT_MIN_SPLIT_MULTIPLIER * (WT_INSERT_KEY_SIZE(ins) + WT_UPDATE_MEMSIZE(ins->upd));
-        if (count > WT_MIN_SPLIT_COUNT && size > (size_t)btree->maxleafpage) {
+        mem_split_threshold = (size_t)WT_MIN(btree->maxleafpage, btree->splitmempage);
```

Comment:
This deserves a comment, something like:

/*
 * Account for the case where the maximum in-memory page size is configured to be smaller than the 
 * maximum on-disk page size. Even with that configuration it is beneficial to be able to in-memory split 
 * for append workloads, and allow the reconciliation to happen in a background thread.
 */

Reply:
@agorrod 
Thanks for your reply, I have increased the disk page limit again, Please review the PR again.

thanks

Reply:
If you are convenient, you can help review this PR:
https://github.com/wiredtiger/wiredtiger/pull/9847

thanks